core/src/epicli/data/common/ansible/playbooks/roles/haproxy/templates/haproxy_RedHat.cfg.j2

```diff
@@ -61,25 +61,37 @@ frontend {{ front.name }}
 {%- for back in specification.backend %}
 backend {{ back.name }}
   balance roundrobin
+  {%- if back.https is defined and back.https == True %}
+  option tcp-check
+  mode tcp
+  {%- endif %}
   {%- if back.server_groups is defined %}
   {%- for server_group in back.server_groups %}
   {%- for server in groups[server_group] %}
+  {%- if back.https is defined and back.https == True %}
+   server {{ server }} {{ hostvars[server]['ansible_default_ipv4']['address'] }}:{{ back.port }} check ssl verify none
+  {%- else %}
    server {{ server }} {{ hostvars[server]['ansible_default_ipv4']['address'] }}:{{ back.port }} check
+  {%- endif %}
   {%- endfor %}
   {%- endfor %}
   {%- endif %}
 
   {%- if back.servers is defined %}
   {%- for server in back.servers %}
-    server {{ server.name }} {{ server.address }}:{{ back.port }} check
+  {%- if back.https is defined and back.https == True %}
+   server {{ server.name }} {{ server.address }}:{{ back.port }} check ssl verify none
+  {%- else %}
+   server {{ server.name }} {{ server.address }}:{{ back.port }} check
+  {%- endif %}
   {%- endfor %}
   {%- endif %}
 {%- endfor %}
 
-    {%- if specification.stats is defined %}
+{%- if specification.stats is defined %}
     {%- if specification.stats.enable %}
 frontend stats
-    mode https
+    mode http
```

Comment:
Why http and no https? 
It will break the sd config for Prometheus in core/core/src/ansible/roles/haproxy_exporter/templates/prometheus-haproxy-exporter.service.j2

Reply:
As this is stated in documentation, there is no such mode as https. 

mode { tcp|http|health }

More ind documentation below:
http://cbonte.github.io/haproxy-dconv/1.8/configuration.html#4.2-mode